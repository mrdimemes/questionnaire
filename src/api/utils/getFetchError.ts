import { AxiosError } from "axios";

import { FetchError, ErrorStatusCode } from "../errors";

const getFetchError = (error: unknown) => {
  if (error instanceof AxiosError && error.response) {
    return new FetchError(
      error.response.status,
      error.response.data.message,
      error.response.data.errors,
    );
  } else if (error instanceof AxiosError && error.request) {
    return new FetchError(
      ErrorStatusCode.ServiceUnavailable,
      "Сервер временно недоступен",
    );
  }
};

export default getFetchError;