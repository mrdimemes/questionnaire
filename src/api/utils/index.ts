export { default as getFetchError } from "./getFetchError";