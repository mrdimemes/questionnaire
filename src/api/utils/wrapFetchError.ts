import { AxiosError } from "axios";

import {
  FetchError,
  ClientError,
  ServerError,
  HTTPErrorStatusCode,
} from "../exceptions";


const wrapFetchError = (error: unknown): FetchError | unknown => {
  if (error instanceof AxiosError && error.response) {
    switch (error.response.status) {
      case HTTPErrorStatusCode.BadRequest:
        return ClientError.BadRequestError(
          error.response.data.message,
          error.response.data.errors,
        );
      case HTTPErrorStatusCode.NotFound:
        return ClientError.NotFoundError(error.response.data.message);
      case HTTPErrorStatusCode.Unauthorized:
        return ClientError.UnauthorizedError();
      case HTTPErrorStatusCode.Forbidden:
        return ClientError.AccessDeniedError();
      case HTTPErrorStatusCode.ServerError:
        return ServerError.ServerError();
    }
  } else if (error instanceof AxiosError && error.request) {
    return ServerError.ServiceUnavailableError();
  } else {
    return error;
  };
};

export default wrapFetchError;