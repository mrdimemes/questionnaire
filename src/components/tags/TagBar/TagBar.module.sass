@use "src/sass/foundation/breakpoints"

.TagBar
  display: flex
  flex-wrap: wrap
  gap: 15px
  @media (max-width: breakpoints.$small-screen)
    gap: 10px
  @media (max-width: breakpoints.$tablet)
    gap: 7px
  @media (max-width: breakpoints.$mobile)
    gap: 5px