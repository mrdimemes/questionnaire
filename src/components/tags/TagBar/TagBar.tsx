import { TagComponent } from "src/components";
import styles from "./TagBar.module.sass";

type TagBarProps = {
  tags: {
    id: number,
    label: string
  }[]
};

const TagBar = ({ tags }: TagBarProps) => {
  return (
    <div className={styles.body}> {
      tags.map((tag) => {
        return <TagComponent key={tag.id} label={tag.label} />
      })
    } </div>
  )
}

export default TagBar