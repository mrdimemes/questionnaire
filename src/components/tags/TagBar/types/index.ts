export type { TagBarProps } from "./TagBarProps";