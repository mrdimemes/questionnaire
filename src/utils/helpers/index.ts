export { default as executeWithConfirmation } from "./executeWithConfirmation";