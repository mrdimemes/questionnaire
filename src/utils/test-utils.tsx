import React, { PropsWithChildren } from "react";
import { BrowserRouter } from "react-router-dom";
import { render, RenderOptions } from "@testing-library/react";
import { PreloadedState } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import { setupStore, AppStore, RootState } from "src/redux";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>{children}</Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

export function renderWithRouters(
  ui: React.ReactElement,
  { ...renderOptions }: ExtendedRenderOptions = {},
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <BrowserRouter>{children}</BrowserRouter>;
  }
  return render(ui, { wrapper: Wrapper, ...renderOptions });
}

export function renderWithProvidersAndRouters(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) {
  function Wrapper({ children }: PropsWithChildren<{}>): JSX.Element {
    return <Provider store={store}>
      <BrowserRouter>{children}</BrowserRouter>
    </Provider>;
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}