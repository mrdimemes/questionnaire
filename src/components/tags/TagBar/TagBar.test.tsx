import { screen } from "@testing-library/react";
import { renderWithProviders } from "src/utils/test-utils";
import TagBar from "./TagBar";

const tags = [
  {
    id: 0,
    label: "Tag1"
  },
  {
    id: 1,
    label: "Tag2"
  },
  {
    id: 3,
    label: "Tag3"
  },
]

describe("TagBar component", () => {
  test("TagBar renders", () => {
    renderWithProviders(<TagBar tags={tags}/>);
    expect(screen.getByText(/tag1/i)).toBeInTheDocument();
    expect(screen.getByText(/tag2/i)).toBeInTheDocument();
    expect(screen.getByText(/tag3/i)).toBeInTheDocument();
  });

  test("TagBar snapshot", () => {
    const { container } = renderWithProviders(<TagBar tags={tags}/>);
    expect(container).toMatchSnapshot();
  });
});