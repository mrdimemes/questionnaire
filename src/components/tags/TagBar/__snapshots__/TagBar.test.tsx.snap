// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagBar component TagBar snapshot 1`] = `
<div>
  <div
    class="body"
  >
     
    <div
      class="body theme_light"
    >
      <div>
        Tag1
      </div>
    </div>
    <div
      class="body theme_light"
    >
      <div>
        Tag2
      </div>
    </div>
    <div
      class="body theme_light"
    >
      <div>
        Tag3
      </div>
    </div>
     
  </div>
</div>
`;
